pkg/cmd/dedicated/deregister/deregister.go

```diff
@@ -0,0 +1,304 @@
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "deregister-cluster",
+		Short:   f.Localizer.MustLocalize("dedicated.registerCluster.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.registerCluster.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.registerCluster.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runDeRegisterClusterCmd(opts)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.clusterManagementApiUrl, "cluster-mgmt-api-url", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterMgmtApiUrl.description"))
+	flags.StringVar(&opts.accessToken, "access-token", "", f.Localizer.MustLocalize("dedicated.registercluster.flag.accessToken.description"))
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("dedicated.registerCluster.flag.clusterId.description"))
+
+	return cmd
+}
+
+func runDeRegisterClusterCmd(opts *options) (err error) {
+	// Set the base URL for the cluster management API
+	err = setListClusters(opts)
+	if err != nil {
+		return err
+	}
+	if len(opts.clusterList) == 0 {
+		return opts.f.Localizer.MustLocalizeError("dedicated.registerCluster.run.noClusterFound")
+	}
+	// TO-DO if client has supplied a cluster id, validate it and set it as the selected cluster without listing getting all clusters
+	if opts.selectedClusterId == "" {
+		err = runClusterSelectionInteractivePrompt(opts)
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range opts.clusterList {
+			cluster := opts.clusterList[i]
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
+	}
+
+	kafkas, err := getkafkasInCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	err = deleteKafkasPrompt(opts, &kafkas)
+	if err != nil {
+		return err
+	}
+
+	err = deregisterClusterFromKasFleetManager(opts)
+	if err != nil {
+		return err
+	}
+
+	err = removeAddonsFromCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getClusterList(opts *options) (*clustersmgmtv1.ClusterList, error) {
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return nil, err
+	}
+	client, cc, err := conn.API().OCMClustermgmt(opts.clusterManagementApiUrl, opts.accessToken)
+	if err != nil {
+		return nil, err
+	}
+	defer cc()
+	// TO-DO deal with pagination, validate clusters -- must be multi AZ and ready.
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return nil, err
+	}
+	clusters := response.Items()
+	return clusters, nil
+}
+
+func setListClusters(opts *options) error {
+	clusters, err := getClusterList(opts)
+	if err != nil {
+		return err
+	}
+	var cls = []clustersmgmtv1.Cluster{}
+	cls = validateClusters(clusters, cls)
+	opts.clusterList = cls
+	return nil
+}
+
+func validateClusters(clusters *clustersmgmtv1.ClusterList, cls []clustersmgmtv1.Cluster) []clustersmgmtv1.Cluster {
```

Comment:
is this needed?

Reply:
this shouldn't be needed, as we are looking to use the name of the cluster (got via OCM) we should cross check the list of clusters with a list of clusters from KFM